### PR TITLE
Connecting shipping rates

### PR DESCRIPTION
**Pull request by atorresveiga**
Part of: #12998

### Description
This PR connects shipping rates with the mock data returned from the `GetShippingRates` use case. Additionally, it adds logic for changing the sort order and refreshing the shipping rates.

Changes included here:
- Display shipping rates from the `GetShippingRates` use case.
- Sort shipping rates using the sort order selected
- Refresh shipping rates (used in the error screen)
- Added a loading state to indicate that the app is working while we are fetching shipping rates

### Testing information
TC1
- Open the app
- Navigate to orders
- Tap on an order eligible for creating a shipping label
- Tap on create a shipping label
- Check that the shipping rates are displayed
- Change the shipping sort order
- Check that shipping rates are refreshed

TC2
First apply this patch to force the get shipping labels use case to fail

<details>
  <summary>Patch</summary>

```
Index: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/GetShippingRates.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/GetShippingRates.kt b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/GetShippingRates.kt
--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/GetShippingRates.kt	(revision 063540b77d705976f703f59726e538865681c073)
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/GetShippingRates.kt	(date 1733926052842)
@@ -5,6 +5,7 @@
 import kotlinx.coroutines.delay
 import javax.inject.Inject
 import kotlin.random.Random
+
 @Suppress("MagicNumber")
 class GetShippingRates @Inject constructor() {
     private val cheapestComparator = Comparator<ShippingRateUI> { r1, r2 ->
@@ -62,13 +63,14 @@
             )
         }
 
-        return Result.success(
+        /*return Result.success(
             carriers.associateWith {
                 generateRates(
                     it.name,
                     Random(0).nextInt(from = 3, until = 10)
                 ).sortedWith(comparator)
             }
-        )
+        )*/
+        return Result.failure(Exception("Random exception"))
     }
 }

```

</details>

- Open the app
- Navigate to orders
- Tap on an order eligible for creating a shipping label
- Tap on create a shipping label
- Check that an error is displayed
- Tap on refresh
- Check that shipping labels refresh (displaying the error again)

### The tests that have been performed
- Testing that the UI works as expected

### Images/gif

https://github.com/user-attachments/assets/c867a224-24ee-4e8e-a15f-1f37de539a77

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md --

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/GetShippingRates.kt [hunk 1]
```diff
@@ -0,0 +1,74 @@
+package com.woocommerce.android.ui.orders.wooshippinglabels
+
+import com.woocommerce.android.R
+import com.woocommerce.android.ui.orders.wooshippinglabels.packages.datasource.PackageDAO
+import kotlinx.coroutines.delay
+import javax.inject.Inject
+import kotlin.random.Random
+@Suppress("MagicNumber")
```
Comment (atorresveiga) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/GetShippingRates.kt:8]: This use case will be updated in the next PR to fetch real data, for now, I'm skipping unit testing it

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/ShippingRatesSection.kt [hunk 2]
```diff
@@ -499,7 +527,7 @@ data class Carrier(
     val logoRes: Int? = null,
 )
 
-data class ShippingRate(
+data class ShippingRateUI(
```
Comment (atorresveiga) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/ShippingRatesSection.kt:530]: Renaming ShippingRate -> ShippingRateUI. The new model will be added in the next PR

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/WooShippingLabelCreationScreen.kt [hunk 3]
```diff
@@ -198,28 +211,67 @@ private fun LabelCreationScreenWithBottomSheet(
...
+                    onSelectedRateSortOrderChanged = onSelectedRateSortOrderChanged,
+                    onRefreshShippingRates = onRefreshShippingRates
                 )
             }
         }
     }
 }
 
+@Composable
+private fun WooShippingShippingRatesSection(
```
Comment (atorresveiga) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/WooShippingLabelCreationScreen.kt:225]: This new composable will be in charge of representing the shipping rate status

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/WooShippingLabelCreationViewModel.kt [hunk 4]
```diff
@@ -26,55 +32,99 @@ class WooShippingLabelCreationViewModel @Inject constructor(
...
     private val navArgs: WooShippingLabelCreationFragmentArgs by savedState.navArgs()
-    private val storeOptions = StoreOptionsModel(
+    private val mockStoreOptions = StoreOptionsModel(
         currencySymbol = "$",
         dimensionUnit = "cm",
         weightUnit = "kg",
         originCountry = "US"
     )
 
+    private val mockSelectedPackage = PackageDAO(
```
Comment (atorresveiga) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/WooShippingLabelCreationViewModel.kt:46]: Shipping rates need a package to be selected

Comment (ThomazFB) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/WooShippingLabelCreationViewModel.kt:46]: I'll adjust this through https://github.com/woocommerce/woocommerce-android/pull/13111.

**Review by ThomazFB — APPROVED** [review]
Looks good, @atorresveiga! I focused on checking the UI and overall wiring between ViewModel and use case. Looks good and great work with the unit test coverage.

**Pull request merged**